src/networking/SocketListener.java

```diff
@@ -0,0 +1,227 @@
+package networking;
+
+import java.io.*;
+import java.net.*;
+
+import networking.queueManagement.ConcurrentBlockingQueue;
+import networking.utility.IncomingPacket;
+
+/**
+ * 
+ * This is the file which contains the SocketListener Class which is a runnable class, that means it
+ * has the functionality of threads. This thread basically keeps listening on a port for the client requests, whenever
+ * there is a request it accepts(In a blocking manner) them and connects them to another socket which basically
+ * receives the message. After receiving the message from the client these messages are 
+ * pushed into either content module queue or processing module queue based on the identifier. 
+ * 
+ * @author Marella Shiva Sai Teja
+ */
+
+public class SocketListener implements Runnable {
+	
+	/**
+	 * Port number on which the socket will be listening for the client requests to connect with.
+	 */
+	int port;
+
+	/**
+	 * socket object which keeps listening for the client requests.
+	 */
+	private ServerSocket serverSocket;
+
+	/**
+	 * Queue into which message from content module will be pushed
+	 */
+	private ConcurrentBlockingQueue<IncomingPacket> contModuleQueue;
+
+	/**
+	 * Queue into which message from processing module will be pushed
+	 */
+	private ConcurrentBlockingQueue<IncomingPacket> procModuleQueue;
+
+	/**
+	 * 
+	 * This method is the constructor of the class which initializes the params
+	 * @param port
+	 * @param contModuleQueue
+	 * @param procModuleQueue
+	 * 
+	 * There won't be any return type as it is a constructor of the class
+	 */
+	public SocketListener(int port, ConcurrentBlockingQueue<IncomingPacket> contModuleQueue, ConcurrentBlockingQueue<IncomingPacket> procModuleQueue){
```

Comment:
Instead of ConcurrentBlockingQueue use IQueue in the arguments of the constructor.

Reply:
IQueue is an interface right.. Internally it calls ConcurrentBlockingQueue, Is it a convention or in future more queues are implemented from that interface it selects the right one? And also should i do it while declaring them also or else only in constructor?